src/handlers/get-file-set-auth.js

```diff
@@ -0,0 +1,65 @@
+const { processRequest, processResponse } = require("./middleware");
+const { getFileSet } = require("../api/opensearch");
+const { isFromReadingRoom } = require("../helpers");
+const isObject = require("lodash.isobject");
+const jwt = require("jsonwebtoken");
+
+/**
+ * Authorizes a FileSet by id
+ */
+exports.handler = async (event) => {
+  event = processRequest(event);
+  const id = event.pathParameters.id;
+  const osResponse = await getFileSet(id, {
+    allowPrivate: true,
+    allowUnpublished: true,
+  });
+
+  if (osResponse.statusCode != 200) {
+    return sendResponse(event, osResponse.statusCode, osResponse.statusCode);
+  }
+
+  const body = JSON.parse(osResponse.body);
+
+  if (!body?.found) {
+    return sendResponse(event, 404, 404);
+  }
+
+  const fileSet = JSON.parse(osResponse.body)._source;
+  const token = event.cookieObject.dcApiV2Token;
+  const visibility = fileSet.visibility;
+  const published = fileSet.published;
+  const readingRoom = isFromReadingRoom(event);
+
+  if (isAllowedVisibility(token, visibility, readingRoom) && published) {
+    return sendResponse(event, 204, "Authorized");
+  } else {
+    return sendResponse(event, 403, "Forbidden");
+  }
+};
+
+function sendResponse(event, statusCode, body) {
+  return processResponse(event, {
+    statusCode: statusCode,
+    body: body,
+  });
+}
+
+function isAllowedVisibility(token, visibility, readingRoom) {
+  switch (visibility) {
+    case "Public":
+      return true;
+    case "Institution":
+      isValidToken(token);
```

Comment:
Missing the `return` keyword.
```suggestion
      return isValidToken(token);
```

Reply:
Always forget the return in JS

Reply:
Same. Ruby and Elixir have ruined us. 😄